scipy/stats/_survival.py

```diff
@@ -374,3 +391,138 @@ def _ecdf_right_censored(sample):
+    malignant gliomas was investigated. The two groups are characterized by
+    a different type of tumor. The week to death of 51 adults was recorded.
+
+    >>> from scipy import stats
+    >>> x = stats.CensoredData(
+    ...     uncensored=[6, 13, 21, 30, 37, 38, 49, 50,
+    ...                 63, 79, 86, 98, 202, 219],
+    ...     right=[31, 47, 80, 82, 82, 149]
+    ... )
+    >>> y = stats.CensoredData(
+    ...     uncensored=[10, 10, 12, 13, 14, 15, 16, 17, 18, 20, 24, 24,
+    ...                 25, 28,30, 33, 35, 37, 40, 40, 46, 48, 76, 81,
+    ...                 82, 91, 112, 181],
+    ...     right=[34, 40, 70]
+    ... )
+
+    We can calculate and visualize the empirical survival functions
+    of both groups as follows.
+
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>> ax = plt.subplot()
+    >>> ecdf_x = stats.ecdf(x)
+    >>> ax.step(np.insert(ecdf_x.x, 0, 0),
+    ...         np.insert(ecdf_x.sf.points, 0, 1),
+    ...         where='post', label='Astrocytoma')
+    >>> ecdf_y = stats.ecdf(y)
+    >>> ax.step(np.insert(ecdf_y.x, 0, 0),
+    ...         np.insert(ecdf_y.sf.points, 0, 1),
+    ...         ls='-.', where='post', label='Glioblastoma')
+    >>> ax.set_xlabel('Time to death (weeks)')
+    >>> ax.set_ylabel('Empirical SF')
+    >>> plt.legend()
+    >>> plt.show()
+
+    Based on a visual inspection of the survival functions, we see a difference
+    but cannot conclude about its significance.
+
+    Next, we will use the logrank test to assess wether the difference
+    between the two survival functions is significant.
+
+    >>> res = stats.log_rank(x=x, y=y)
+    >>> res.statistic
+    7.49659416854
+    >>> res.pvalue
+    0.006181578637
+
+    Using a significance level of 5%, we would reject the null hypothesis in
+    favor of the alternative hypothesis: "there is a difference between the
+    two survival functions".
+
+    """
+    x = _iv_CensoredData(sample=x, param_name='x')
+    y = _iv_CensoredData(sample=y, param_name='y')
+
+    xy = CensoredData(
+        uncensored=np.concatenate((x._uncensored, y._uncensored)),
+        right=np.concatenate((x._right, y._right))
+    )
+
+    res = ecdf(xy)
+    idx = res.sf._d.astype(bool)  # events were observed (not censored)
+    times_xy = res.x[idx]
+    at_risk_xy = res.sf._n[idx]
+    deaths_xy = res.sf._d[idx]
+
+    n_died_x = x._uncensored.size
+
+    res_x = ecdf(x)
+    i = np.searchsorted(res_x.x, times_xy)  # or use `interpolate_1d`
+    at_risk_x = np.append(res_x.sf._n, 0)[i]  # 0 at risk after last x time
+    at_risk_y = at_risk_xy - at_risk_x
+
+    sum_exp_deaths_x = np.sum(at_risk_x * (deaths_xy/at_risk_xy))
+
+    # equivalent: formulation with the mean
+    # statistic = (
+    #     (n_died_x - sum_exp_deaths_x)**2/sum_exp_deaths_x
+    #     + (n_died_y - sum_exp_deaths_y)**2/sum_exp_deaths_y
+    # )
+    # warning occurs when multiple deaths at same time
```

Comment:
I don't think that's when this occurs.
```suggestion
    # warning occurs when `at_risk_xy == 1`. Effectively, we have 
    # `(at_risk_xy -1)` in the numerator and denominator. Simplifying these
    # expressions symbolically, we would always find the term to be zero.
```

Reply:
Correct, was explanation was quick and wrong 👍